gnomad_hail/utils/annotations.py

```diff
@@ -419,3 +425,503 @@ def annotate_freq(
+
+    The sibling sharing counts can be stratified using additional filters using `stata`.
+
+    .. note::
+
+        This function expects the MT has either been split or filtered to only bi-allelics
+
+    :param sib_mt: Input sibling Matrix table generated by `create_sib_mt`
+    :param strata: Dict with additional strata to use when computing shared sibling variant counts
+    :param sib1_entry: Entry containing the 1st sibling of the pair in sibling MT
+    :param sib2_entry: Entry containing the 2nd sibling of the pair in sibling MT
+    :param sib1_is_female_col: An optional column in sib_mt giving sib2 sex. If not given, counts are only computed for autosomes.
+    :param sib2_is_female_col: An optional column in sib_mt giving sib2 sex. If not given, counts are only computed for autosomes.
+    :return: A Table with the sibling shared variant counts
+    """
+
+    def get_alt_count(locus, gt, is_female):
+        """
+        Helper method to calculate alt allele count with sex info if present
+        """
+        return (
+            hl.case()
+            .when(
+                ~is_female & (locus.in_x_nonpar() | locus.in_y_nonpar()),
+                hl.min(1, gt.n_alt_alleles()),
+            )
+            .when(is_female & locus.in_y_nonpar(), 0)
+            .when(is_female | locus.in_autosome_or_par(), gt.n_alt_alleles())
+            .default(0)
+        )
+
+    # Create sibling sharing counters
+    sib_stats = hl.struct(
+        **{
+            f"n_sib_shared_variants_{name}": hl.agg.filter(
+                expr,
+                hl.or_missing(
+                    ((sib1_is_female_col is not None) & (sib2_is_female_col is not None)) | sib_mt.locus.in_autosome(),
+                    hl.agg.sum(
+                        hl.if_else(
+                            hl.is_missing(sib_mt[sib1_entry].GT.n_alt_alleles()) | hl.is_missing(sib_mt[sib2_entry].GT.n_alt_alleles()),
+                            0,
+                            hl.min(
+                                get_alt_count(
+                                    sib_mt.locus, sib_mt[sib1_entry].GT, sib_mt[sib1_is_female_col]
+                                ),
+                                get_alt_count(
+                                    sib_mt.locus, sib_mt[sib2_entry].GT, sib_mt[sib2_is_female_col]
+                                ),
+                            ),
+                        )
+                    ),
+                ),
+            )
+            for name, expr in strata.items()
+        }
+    )
+
+    sib_stats = sib_stats.annotate(
+        **{
+            f'ac_sibs_{name}': hl.agg.filter(
+                expr,
+                hl.agg.sum(sib_mt.sib1_entry.GT.n_alt_alleles() + sib_mt.sib2_entry.GT.n_alt_alleles())
+            ) for name, expr in strata.items()
+        }
+    )
+
+    return sib_stats
+
+
+def create_sib_mt(
```

Comment:
Is this sibs MT used for any other purpose than creating those sibs counts?  If  not, this seems like an awful lot of work that I don't think is necessary. If you only created the `sibs` global annotation, it seems like that'd be  enough to compute these stats without reorganizing the whole MT.